docs/src/generative.md

```diff
@@ -2,48 +2,101 @@
 
 ## Description
 
-The *generative* interface contains a small collection of functions that makes implementing and solving problems with generative models easier. These functions return states and observations instead of distributions as in the [Explicit interface](@ref explicit_doc).
+The *generative* interface consists of two functions:
+- [`gen`](@ref) returns samples (e.g. states, observations and rewards) from a generative POMDP model.
+- [`initialstate`](@ref) returns a sampled initial state.
+The generative interface is typically used when it is easier to return sampled states and observations rather than explicit distributions as in the [Explicit interface](@ref explicit_doc).
+This type of model is often referred to as a "black-box" model.
 
-The generative interface functions are the following (note that this is not actual julia code):
-```julia
-generate_s(pomdp, s, a, rng) -> sp
-generate_o(pomdp, s, a, sp, rng) -> o
-generate_sr(pomdp, s, a, rng) -> (s, r)
-generate_so(pomdp, s, a, rng) -> (s, o)
-generate_or(pomdp, s, a, sp, rng) -> (o, r)
-generate_sor(pomdp, s, a, rng) -> (s, o, r)
-initialstate(pomdp, rng) -> s
+## Basic Implementation Instructions
+
+To implement the generative interface for a 'POMDP' or 'MDP' type 'M', one should implement
+
+- [`gen`](@ref)`(m::M, s, a, `[`rng`](@ref Random-number-generators)`)`. This should return a [`NamedTuple`](todo link) with entries `sp` (mnemonic "s-prime") for the next state, `r` for the reward, and `o` for the observation (if `M <: POMDP`).
```

Comment:
Also, this sentence sounds like `gen(m, s, a, rng)` *has* to return the entire output tuple. However, as you stated below, this might be an arbitrary subset.

Reply:
> might be preferred because it allows to automatically create the versions that with subsets of the full sor-tuple.

These versions will still be created if `gen(m, s, a, rng)` is implemented (they just might do some extra computation)